src/internal.c

```diff
@@ -10663,7 +10663,8 @@ int ProcessPeerCerts(WOLFSSL* ssl, byte* input, word32* inOutIdx,
             args->count = args->totalCerts;
             args->certIdx = 0; /* select peer cert (first one) */
 
-            if (args->count == 0 && ssl->options.mutualAuth &&
+            if (args->count == 0 && (ssl->options.mutualAuth ||
+                 (ssl->options.failNoCert && IsAtLeastTLSv1_3(ssl->version))) &&
```

Comment:
This doesn't look right to me.  That is, TLS 1.3 shouldn't have a separate test.  We should be failing if failNoCert is on independent of protocol version.  How come failNoCert wasn't checked before here?

Reply:
failNoCert is set by the wolfSSL_CTX_set_verify().
I want the TLS 1.2 behaviour to be maintained - fail in Key Exchange.
But I want TLS 1.3 to fail when ProcessingPeerCerts.
When wolfSSL_CTX_set_verify() is called we don't know which protocol is going to be negotiated.
So, I have to check the failNoCert flag in ProcessPeerCerts, when we know the negotiated protocol version.
That's how we ended up like this.